File: microservices/store_sensor_api/src/main/kotlin/com/example/sensor_api/domain/models/SensorModel.kt
package com.example.sensor_api.domain.models

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.databind.PropertyNamingStrategies
import com.fasterxml.jackson.databind.annotation.JsonNaming
import io.quarkus.mongodb.panache.common.MongoEntity

@MongoEntity(collection="guitar_sensor_data")
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy::class)
data class SensorModel(
    var sensor_type: String = "",
    var manufacturer: String = "",
    var model: String = "",
    var serial_number: String = "",
    var timestamp: String = "",
    var notes_per_minute: Int = 0,
    var loudness: Int = 0,
    @field:JsonProperty("string_vibration_amplitude")
    var stringVibrationAmplitude: StringVibrationAmplitude? = StringVibrationAmplitude(),
)

File: gateways/web_gateway/src/models/SensorModel.ts
import StringVibrationAmplitude from "./StringVibrationAmplitude";

class SensorModel {
  sensor_type?: string;
  manufacturer?: string;
  model?: string;
  serial_number?: string;
  timestamp?: string;
  notes_per_minute: number | undefined;
  loudness: number | undefined;
  string_vibration_amplitude?: StringVibrationAmplitude;
}

export default SensorModel;
